src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -68,16 +68,10 @@ public void BeginRequest(HttpContext httpContext, ref HostingApplication.Context
             // To avoid allocation, return a null scope if the logger is not on at least to some degree.
             if (loggingEnabled)
             {
-                // Get the request ID (first try TraceParent header otherwise Request-ID header
-                if (!httpContext.Request.Headers.TryGetValue(TraceParentHeaderName, out var correlationId))
-                {
-                    httpContext.Request.Headers.TryGetValue(RequestIdHeaderName, out correlationId);
-                }
-
                 // Scope may be relevant for a different level of logging, so we always create it
                 // see: https://github.com/aspnet/Hosting/pull/944
                 // Scope can be null if logging is not on.
-                context.Scope = _logger.RequestScope(httpContext, correlationId);
+                context.Scope = _logger.RequestScope(httpContext, context.Activity.Id);
```

Comment:
@SergeyKanzhelev @lmolkova should we split this out into span id and trace id in the logs?

Reply:
This is what loggers are typically doing. Like in OpenCensus: https://github.com/census-instrumentation/opencensus-specs/blob/c3fd41b28fed96ceffc8c35af280d74cacea3ab2/trace/LogCorrelation.md#key-names-for-tracing-data or Elastic: https://github.com/elastic/ecs/issues/99